### PR TITLE
[lexical-playground][lexical-table] Bug Fix: Resizing table with merged cells

### LINKED ISSUE #5515: Bug: This is a bug where the table width cannot be adjusted after merging cells.
Lexical version: **2024-01-19 playground version**

## Steps To Reproduce

1. Add a table.
2. Merge cells.
3. Adjust the width of the last column.
4. I can't control the screen.

**### Attached is a recorded video.**
## The current behavior
https://github.com/facebook/lexical/assets/157095195/e8b1687e-8e47-408f-8a2d-dd9f45b3c863

### PR DESCRIPTION
**Pull request by AlexanderReznik**
### Changing the selection of cells that needs to be resized during resizing of a column
Current logic works nicely for the cases without merged cells, however it breaks after cells merging. I'm using `$computeTableMapSkipCellCheck` to create a mapping of cells to their coordinates on screen on every resize. And I think it might be beneficial to cache the map in the `TableNode` as it is useful for selection, resizing, navigation in a table with keyboard (that currently is also crashing with the merged cells). I'd appreciate the thoughts on whether it's possible to extend the `TableNode`.

### Changing the logic for calculating new width/height of cells
Assuming that the new height/width of the cell that is active during the resize should be applied to all cells in the same row/column is incorrect because after merging cells in a table cells in the same row/column can have different height/width. This means the change in height/width should be applied to every cell that is resized.

---

Closes #5515 

## Test plan
I'm currently adding tests for the cases where resizing was breaking previously.

### Before
Crash:
resize-crash

Wrong size:
resize-wrong-width

### After
Right size:
resize-wrong-width-after

### CODE REVIEW

File: packages/lexical-playground/src/plugins/TableCellResizer/index.tsx [hunk 1]
```diff
@@ -199,48 +256,31 @@ function TableCellResizer({editor}: {editor: LexicalEditor}): JSX.Element {
...
             ) {
-              throw new Error('Expected table cell to be inside of table row.');
-            }
-
-            const tableCell = rowCells[rowColumnIndexWithSpan];
-
-            if (!$isTableCellNode(tableCell)) {
-              throw new Error('Expected table cell');
+              const width = getCellNodeWidth(cell.cell, editor);
+              if (!width) {
```
Comment (potatowagon) [inline packages/lexical-playground/src/plugins/TableCellResizer/index.tsx:278]:
```suggestion
              if (width === undefined) {
```

**Review by potatowagon — DISMISSED** [review]
great fix thankyou!

File: packages/lexical-playground/src/plugins/TableCellResizer/index.tsx [hunk 2]
```diff
@@ -178,6 +182,17 @@ function TableCellResizer({editor}: {editor: LexicalEditor}): JSX.Element {
             throw new Error('Expected table row');
           }
 
+          let height = tableRow.getHeight();
+          if (!height) {
```
[the remark this answers is not in the document]
Comment (AlexanderReznik) [reply packages/lexical-playground/src/plugins/TableCellResizer/index.tsx]: Yes, makes sense!
Changed to more specific checks in https://github.com/facebook/lexical/pull/6200/commits/9fa9e0fed20012ca285f1070be1f538187398da5

**Review by potatowagon — APPROVED** [review]

File: packages/lexical-playground/src/plugins/TableCellResizer/index.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (potatowagon) [reply packages/lexical-playground/src/plugins/TableCellResizer/index.tsx]: thankyou for the updates!

**Comment by Sahejkm** [comment]
Hi @AlexanderReznik  this PR is actually failing the CI , the newly added e2e tests written are failing ,

example: https://github.com/facebook/lexical/actions/runs/9311543149/job/25630871373?pr=6216

**Pull request merged**